app/src/main/java/bg/zahov/app/ui/settings/SettingsScreen.kt

```diff
@@ -0,0 +1,282 @@
+package bg.zahov.app.ui.settings
+
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonColors
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.lifecycle.viewmodel.compose.viewModel
+import bg.zahov.app.data.local.RealmManager.Companion.AUTOMATIC_SYNC_SETTING
+import bg.zahov.app.data.local.RealmManager.Companion.LANGUAGE_SETTING
+import bg.zahov.app.data.local.RealmManager.Companion.REST_TIMER_SETTING
+import bg.zahov.app.data.local.RealmManager.Companion.SOUND_EFFECTS_SETTING
+import bg.zahov.app.data.local.RealmManager.Companion.SOUND_SETTING
+import bg.zahov.app.data.local.RealmManager.Companion.THEME_SETTING
+import bg.zahov.app.data.local.RealmManager.Companion.UNIT_SETTING
+import bg.zahov.app.data.local.RealmManager.Companion.UPDATE_TEMPLATE_SETTING
+import bg.zahov.app.data.local.RealmManager.Companion.VIBRATION_SETTING
+import bg.zahov.app.data.local.RealmManager.Companion.WATCH_SETTINGS
+import bg.zahov.app.data.model.LanguageKeys.Companion.languages
+import bg.zahov.app.data.model.SoundKeys.Companion.sounds
+import bg.zahov.app.data.model.ThemeKeys.Companion.theme
+import bg.zahov.app.data.model.UnitsKeys.Companion.units
+import bg.zahov.fitness.app.R
+
+@Composable
+fun SettingsScreen(
+    viewModel: SettingsViewModel = viewModel(),
+    navigateBack: () -> Unit,
+    navigateEditProfile: () -> Unit
+) {
+    val uiState by viewModel.uiState.collectAsStateWithLifecycle()
+    val context = LocalContext.current
+
+    if (uiState.returnBack) {
+        navigateBack()
```

Comment:
Whenever you have something that change the state of the program, use Effects, otherwise you will experience strange behavior like flickering, due to calling some function on every recomposition which not change ui state